### PR TITLE
Filter Ignorable Copyrights #372

### PR DESCRIPTION
**Pull request by adityaviki**
A post-scan plugin to filter out ignorable-copyrights that are read from license's yaml file.

### CODE REVIEW

**Comment by adityaviki** [comment]
@pombredanne let me know if I need to make any improvements to the code. I'll write the the tests for it afterwards.

**Comment by sritasngh** [comment]
Can you please edit the top comment and linked issue no along with your name! This makes easy to understand in that way:)
You could see [here](https://github.com/nexB/scancode-toolkit/pull/2433#issue-591072762) the template. Thank You for your contribution:)

File: docs/source/cli-reference/help-text-options.rst [hunk 1]
```diff
@@ -104,6 +104,8 @@ The Following Help Text is displayed, i.e. This is the help text for Scancode Ve
                                       Files and directories without findings
                                       are omitted (file information is not
                                       treated as findings).
+      --filter-ignoralbe-copyrights   Filter out copyrights that are for the
```
Comment (pombredanne) [inline docs/source/cli-reference/help-text-options.rst]: I think we have a tiny typo:
`--filter-ignoralbe-copyrights` should be `--filter-ignorable-copyrights`
```suggestion
      --filter-ignorable-copyrights   Filter out copyrights that are for the 
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you ++
This is looking great! :bow: 
It is missing one thing though before I can merge: some test.
Do you think you can draft one?

**Comment by adityaviki** [comment]
> Do you think you can draft one?

Yes of course :)

**Comment by adityaviki** [comment]
> Can you please edit the top comment and linked issue no along with your name!

Sorry! I didn't understand. Do I need to add

**Comment by adityaviki** [comment]
I didn't mean to close the PR or delete the branch. I wasn't  paying  attention to where I was clicking. :/

File: src/cluecode/plugin_filter_ignorable_copyrights.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+from commoncode.cliutils import OUTPUT_FILTER_GROUP
```
Comment (pombredanne) [inline src/cluecode/plugin_filter_ignorable_copyrights.py:10]: Could add a header comment matching other files?

**Review by pombredanne — CHANGES_REQUESTED** [review]

**Comment by pombredanne** [comment]
> I didn't mean to close the PR or delete the branch. I wasn't paying attention to where I was clicking. :/

that's OK :)

File: src/cluecode/plugin_filter_ignorable_copyrights.py (hunk 2 above)
Comment (pombredanne) [inline src/cluecode/plugin_filter_ignorable_copyrights.py:10]:
```suggestion
#
# Copyright (c) nexB Inc. and others. All rights reserved.
# ScanCode is a trademark of nexB Inc.
# SPDX-License-Identifier: Apache-2.0
# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
# See https://github.com/nexB/scancode-toolkit for support or download.
# See https://aboutcode.org for more information about nexB OSS projects.
#

from commoncode.cliutils import OUTPUT_FILTER_GROUP
```

**Review by pombredanne — APPROVED** [review]

**Comment by adityaviki** [comment]
@pombredanne I don't know how to solve the failing check.

**Comment by sritasngh** [comment]
>I don't know how to solve the failing check.

DCO check is falling because of this https://github.com/nexB/scancode-toolkit/pull/2436/commits/55cea2679297d4f7cd757c8a922b1c0f32906c47 It has co-authour name but signed-off is missing. So what you can do is squash these changes to the original commit of your and add sign-off or you may ask @pombredanne to squash all the commits into one.

**Comment by adityaviki** [comment]
@itssingh It's still not clear how to solve the issue.

**Comment by pombredanne** [comment]
@adityaviki this will be merged at last in v32.1 ... but it may be partially redundant with pre-existing ScanCode CLI options

**Comment by pombredanne** [comment]
https://github.com/nexB/scancode-toolkit/blob/develop/src/cluecode/plugin_filter_clues.py has the feature and predates this PR. Thank you +++, closing.

**Pull request closed without merging**